ludwig/visualize.py

```diff
@@ -147,6 +180,97 @@ def _validate_output_feature_name_from_test_stats(
     except TypeError:
         return output_feature_names_set
 
+def _encode_categorical_feature(
+        raw: np.array,
+        str2idx: dict
+) -> np.array:
+    """encodes raw categorical string value to encoded numeric value
+
+    Args:
+    :param raw: (np.array) string categorical representation
+    :param str2idx: (dict) dictionary that maps string representation to
+        encoded value.
+
+    Returns:
+        np.array
+    """
+    return str2idx[raw]
+
+def _extract_ground_truth_values(
+        ground_truth: str,
+        output_feature_name: str,
+        ground_truth_split: int,
+        ground_truth_metadata: str,
+        split_file: Union[str, None]) -> np.ndarray:
```

Comment:
the return should not be a numpy array, but a pandas column i guess

Reply:
I'd like to confirm the intent of the guidance.

I can make the change to return the pandas column.  This means that I'll have to move the encoding of string values to elsewhere.  I believe we will still need encoded numeric values.  Many of lower level visualization functions work on the encoded values, such as [here](
https://github.com/uber/ludwig/blob/0644620329aef02a4a9517386bc7839fd0192620/ludwig/visualize.py#L1079-L1103)
and [here](https://github.com/uber/ludwig/blob/0644620329aef02a4a9517386bc7839fd0192620/ludwig/visualize.py#L1165-L1198)

If the intent of the guidance is to move the translation of string to encode value outside of the `_extract_ground_truth_values()` function, I can make that change.  It just means duplicating a few lines of code that translates the string values to the encoded values wherever ground truth is needed.

OTOH, if the intent is to have the visualizations work only on the pandas raw string values, then this means more extensive changes are needed elsewhere in the visualization processing, which I'm not sure on the benefit.

Reply:
To make the single viz functions more reusable I'd probably make them accept pandas columns, and change their logic accordingly. I actually believe it would clean them and make tem simpler as it would not require mapping and unmapping values. And in the worst case scenario, one could just do the cast to numpy at the beginning of those functions. What about trying to do this for one function only, the compare performance from probs one for instance, and that would give us some indication if it's worth doing it for the other ones?

Reply:
made modification.

Reply:
I have a first working version of the modified viz functions that work with pandas columns directly.
Here is the modified [`compare_classifiers_performance_from_prob()`](https://github.com/jimthompson5802/ludwig/blob/fcba7649b476369796d7d3518e327603e4382617/ludwig/visualize.py#L1147-L1266).  This contains the fix for the inconsistency issues that I noted on the `master` branch.
This is all remains in [`compare_classifiers_performance_from_prob_cli()`](https://github.com/jimthompson5802/ludwig/blob/fcba7649b476369796d7d3518e327603e4382617/ludwig/visualize.py#L282-L324).  This is now just a minimal wrapper function now.  Maybe we can eliminate this.

Let me know what you think.  If this looks good, I'll do the same with the remaining viz functions.